shared/feature/src/commonMain/kotlin/app/futured/kmptemplate/feature/ui/profileScreen/ProfileScreenNavigation.kt

```diff
@@ -2,6 +2,7 @@ package app.futured.kmptemplate.feature.ui.profileScreen
 
 import app.futured.arkitekt.decompose.navigation.NavigationActions
 
-internal data class ProfileScreenNavigation(
-    val toLogin: () -> Unit,
-) : NavigationActions
+internal interface ProfileScreenNavigation : NavigationActions {
+    fun ProfileScreen.toLogin()
```

Comment:
I would consider naming all navigation extension functions with `navigate` prefix, so this would become 
```suggestion
    fun ProfileScreen.navigateToLogin()
```

That way it will be easier to find the navigation function using autocomplete when working on Component logic.